### PR TITLE
Fix issues relating to loading of invalid data from file

### PR DESCRIPTION
**Pull request by chooyikai**
List of fixes:
1) Attempting to load a Module with null moduleCode or Task with null taskName will abort loading from that file.
2) Attempting to load a Task with 0 or negative taskDuration will set it to null.
3) Attempting to load a Module with null grade will set it to NOT_ENTERED.
4) Attempting to load a config file containing invalid config options will abort loading of settings.

Things to note:
- If a boolean gets corrupted in the save file, it defaults to false. (isTaskDone)
- If an integer gets corrupted in the save file, it defaults to 0. (module credits)

I don't think there's any way we can intercept these because the GSON library does not raise any errors if the JSON is still valid, even if the keys are wrong.

### CODE REVIEW

File: src/main/java/seedu/duke/storage/ModHappyStorageManager.java [hunk 1]
```diff
@@ -69,6 +76,15 @@ public static Configuration loadConfiguration(String configurationPath) {
             modHappyStorage = new ConfigurationStorage();
             try {
                 Configuration configuration = (Configuration) modHappyStorage.loadData(configurationPath);
+                HashMap<Configuration.ConfigurationGroup,String> configMap = configuration.configurationGroupHashMap;
+                for (Configuration.ConfigurationGroup key : configMap.keySet()) {
+                    if (key == null) {
+                        throw new InvalidConfigurationException();
```
Comment (heekit73098) [inline src/main/java/seedu/duke/storage/ModHappyStorageManager.java]: Im not sure if you want to refactor this as it looks like arrowhead style

File: src/main/java/seedu/duke/storage/ModHappyStorageManager.java [hunk 2]
```diff
@@ -110,13 +140,20 @@ public static void loadModuleList(ModuleList moduleList, String modulePath) {
                 ArrayList<Module> arrayListModule;
                 arrayListModule = (ArrayList<Module>) modHappyStorage.loadData(modulePath);
                 for (Module m : arrayListModule) {
+                    if (Objects.isNull(m.getModuleCode())) {
```
Comment (heekit73098) [inline src/main/java/seedu/duke/storage/ModHappyStorageManager.java]: same here, looks like arrowhead style

**Review by heekit73098 — COMMENTED** [review]
Just a suggestion on the code

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -17,11 +17,12 @@
...
    <br>6.2. [Tag Feature](#62-tag-feature)
    <br>6.3. [Grade Feature](#63-grade-feature)
-   <br>6.2. [GPA Feature](#64-gpa-feature) 
-7. [User Stories](#7-user-stories)
-8. [Non-Functional Requirements](#8-non-functional-requirements)
-9. [Glossary](#9-glossary)
-10. [Instructions for manual testing](#10-instructions-for-manual-testing)
+   <br>6.4. [GPA Feature](#64-gpa-feature)
+   <br>6.5. [Storage Feature](#65-storage-feature)
+8. [User Stories](#7-user-stories)
```
Comment (heekit73098) [inline docs/DeveloperGuide.md]: 7 is missing in the index here

Comment (chooyikai) [reply docs/DeveloperGuide.md]: Oops good catch

**Pull request merged**